### PR TITLE
[RV64_DYNAREC] Added more opcodes and a small fix

### PR DESCRIPTION
**Pull request by ksco**
for Terraria.

### CODE REVIEW

File: src/dynarec/rv64/dynarec_rv64_66.c [hunk 1]
```diff
@@ -702,6 +702,26 @@ uintptr_t dynarec64_66(dynarec_rv64_t* dyn, uintptr_t addr, uintptr_t ip, int ni
...
+                        MOV32w(x4, 1);
+                    } else {
+                        INST_NAME("SHL Ew, CL");
+                        ANDI(x4, xRCX, 0x1f);
+                    }
+                    UFLAG_IF {MESSAGE(LOG_DUMP, "Need Optimization for flags\n");}
+                    SETFLAGS(X_ALL, SF_PENDING);
+                    GETEW(x1, 0);
+                    UFLAG_OP12(ed, x4)
+                    SLL(ed, ed, x4);
```
Comment (ptitSeb) [inline src/dynarec/rv64/dynarec_rv64_66.c:718]: you can use option `rv64_zbb` path here, with `zexth(...)`

Comment (ksco) [reply src/dynarec/rv64/dynarec_rv64_66.c:718]: indeed

Comment (ksco) [reply src/dynarec/rv64/dynarec_rv64_66.c:706]: Done, I searched and replaced all `SLLI + SRLI` pairs.

Comment (ptitSeb) [reply src/dynarec/rv64/dynarec_rv64_66.c:706]: Oh, wait, you have to make it optionnal. Zbb is not always supported. Use the `if(rv64_zbb)` Or change the `ZEXTH(...)` opcode itself to check for zbb and use SLLi/SRLI as a fallback.

**Comment by ptitSeb** [comment]
Ah yes, nice! thanks.

File: src/dynarec/rv64/dynarec_rv64_660f.c [hunk 2]
```diff
@@ -419,6 +418,41 @@ uintptr_t dynarec64_660F(dynarec_rv64_t* dyn, uintptr_t addr, uintptr_t ip, int
...
+                    d1 = fpu_get_scratch(dyn);
+                    v1 = fpu_get_scratch(dyn);
+                    MOV64x(x3, 1ULL << __DBL_MANT_DIG__);
+                    FCVTDL(d1, x3, RD_RTZ);
+                    for (int i=0; i<2; ++i) {
+                        FLD(d0, wback, fixedaddress+8*i);
+                        FEQD(x4, d0, d0);
+                        BNEZ_MARK(x4);
+                        B_MARK3_nocond;
+                        MARK; // d0 is not nan
```
Comment (ksco) [inline src/dynarec/rv64/dynarec_rv64_660f.c:437]: @ptitSeb  Just to be sure, is `MARK` macros going to work inside a for loop?

Comment (ptitSeb) [reply src/dynarec/rv64/dynarec_rv64_660f.c:437]: Ah no, it will not. It will just get the MARK for the last iteration

Comment (ksco) [reply src/dynarec/rv64/dynarec_rv64_660f.c:437]: It's computed and applied in different passes, right?

Comment (ptitSeb) [reply src/dynarec/rv64/dynarec_rv64_660f.c:437]: Yes. It's computed on pass2 and applied on pass3.

Comment (ksco) [reply src/dynarec/rv64/dynarec_rv64_660f.c:437]: Ok, I'll fix this in the upcoming PR.

**Pull request merged**